include/particles/particle.tcc

```diff
@@ -872,14 +872,61 @@ void mpm::Particle<Tdim>::compute_strain(double dt) noexcept {
 
 // Compute stress
 template <unsigned Tdim>
-void mpm::Particle<Tdim>::compute_stress() noexcept {
+void mpm::Particle<Tdim>::compute_stress(double dt,
+                                         mpm::StressRate stress_rate) noexcept {
   // Check if material ptr is valid
   assert(this->material() != nullptr);
-  // Calculate stress
-  this->stress_ =
+
+  // Compute material part of stress
+  const Eigen::Matrix<double, 6, 1> material_part_Voigt =
```

Comment:
```suggestion
  const Eigen::Matrix<double, 6, 1>& material_part_Voigt =
```

Reply:
For constant arrays/matrices, using reference is better so you don't copy to a new object.